### ISSUE TITLE
Add a check that CloudWatch Events Rules have <= 5 targets

### ISSUE THREAD

**Issue by adamchainz** [awslabs/cfn-python-lint#306]
*cfn-lint version: (`cfn-lint --version`) 0.5.0* (but nothing changed since afaict)

*Description of issue.*

CloudWatch Events has a limit of 10 targets per Rule, but this isn't validated by `ValidateTemplate` or `cfn-lint`'s rules. Consequently one has to wait til deploy time to discover this error message:

> CREATE_FAILED: At most 10 targets allowed.

Failing template:

```yaml
Resources:
  MyCronRule:
    Type: AWS::Events::Rule
    Properties:
      ScheduleExpression: cron(0 6 * * ? *)
      Targets:
      - Id: Job1
        Arn: arn:target
      - Id: Job2
        Arn: arn:target
      - Id: Job3
        Arn: arn:target
      - Id: Job4
        Arn: arn:target
      - Id: Job5
        Arn: arn:target
      - Id: Job6
        Arn: arn:target
      - Id: Job7
        Arn: arn:target
      - Id: Job8
        Arn: arn:target
      - Id: Job9
        Arn: arn:target
      - Id: Job10
        Arn: arn:target
      - Id: Job11
        Arn: arn:target
```

**Comment by adamchainz** [comment]
I'll also take a look at a PR myself if I find the time

**Comment by adamchainz** [comment]
According to https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/cloudwatch_limits_cwe.html the limit is actually 5 targets per rule. At 8 targets I get a different error message to at 11 targets:

```
$ aws events put-targets --rule test-limits --targets '[
    {"Id": "Test1", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test2", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test3", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test4", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test5", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test6", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"}
]'

An error occurred (LimitExceededException) when calling the PutTargets operation: The requested resource exceeds the maximum number allowed.
$ aws events put-targets --rule test-limits --targets '[
    {"Id": "Test1", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test2", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test3", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test4", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test5", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test6", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test7", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test8", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test9", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test10", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
    {"Id": "Test11", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"}
]'

An error occurred (ValidationException) when calling the PutTargets operation: At most 10 targets allowed.
```

This might imply the ability to use 6-10 targets is possible with a limit increase, although such a limit increase isn't documented. I've reported to support.

**Comment by adamchainz** [comment]
Support confirmed there's no limit increase possible, just a weirdness of the API that it returns two different error messages at different numbers of targets. So a hardcoded limit of 5 would be fine in `cfn-lint`.

**Comment by kddejong** [comment]
Thanks for checking that should make things easier.  It would have been an odd rule if it was a soft limit.

**Comment by VardyNg** [comment]
> According to https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/cloudwatch_limits_cwe.html the limit is actually 5 targets per rule. At 6 targets I get a different error message to at 11 targets:
> 
> ```
> $ aws events put-targets --rule test-limits --targets '[
>     {"Id": "Test1", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test2", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test3", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test4", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test5", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test6", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"}
> ]'
> 
> An error occurred (LimitExceededException) when calling the PutTargets operation: The requested resource exceeds the maximum number allowed.
> $ aws events put-targets --rule test-limits --targets '[
>     {"Id": "Test1", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test2", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test3", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test4", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test5", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test6", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test7", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test8", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test9", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test10", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"},
>     {"Id": "Test11", "Arn": "arn:aws:lambda:us-east-1:123456789012:function:example"}
> ]'
> 
> An error occurred (ValidationException) when calling the PutTargets operation: At most 10 targets allowed.
> ```
> 
> This might imply the ability to use 6-10 targets is possible with a limit increase, although such a limit increase isn't documented. I've reported to support.

The maximum target is still 5 and is not adjustable [Doc](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-quota.html#eb-limits). 

The "At most 10 targets allowed" error is because the request exceeded the maximum rule per request, which is 10
